### ISSUE TITLE
release for ROS kinetic

### ISSUE THREAD

**Issue by wjwwood** [OctoMap/octomap#114]
ROS kinetic is spinning up now and it looks like octomap could be released now. Is it possible to get that done in the near future? I can help if it would help.

Thanks!

**Comment by ahornung** [comment]
Sure, that should be possible! I just haven't had a chance to set up a proper prerelease environment to test. The latest release (1.7.0 in jade) should not cause any problems, could you run a test?

Could octovis cause a problem because it's using libQGLviewer with Qt4 (not yet tested with Qt5)?

Is there a definite date for the kinetic release? There could be some API-breaking changes in the next OctoMap release. The best time frame for introducing them will probably be now or *soon*.

**Comment by wjwwood** [comment]
> Sure, that should be possible! I just haven't had a chance to set up a proper prerelease environment to test. The latest release (1.7.0 in jade) should not cause any problems, could you run a test?

The best thing to do is to just release the jade version into Kinetic and see if it works, imo. If it doesn't you'll have to make a fix, at which point you'll need to decide if that fix should be applied to jade as well as kinetic or not and that will determine if you need to have a different version in each.

> Could octovis cause a problem because it's using libQGLviewer with Qt4 (not yet tested with Qt5)?

So rviz uses Qt5 (or will) in Kinetic, but Gazebo7 will use Qt4. This appears to be fine (Qt4 and Qt5 do not conflict with each other). If you have an rviz plugin that uses libQGLViewer then that would be a problem.

> Is there a definite date for the kinetic release? There could be some API-breaking changes in the next OctoMap release. The best time frame for introducing them will probably be now or soon.

We're shooting for World Turtle Day (May 23). Now would definitely be the right time to introduce those changes.

**Comment by wjwwood** [comment]
I can do the release if it helps you out, but it should just be running `bloom-release`.

**Comment by ahornung** [comment]
OK, I'll give it a try!

**Comment by ahornung** [comment]
It seems like octomap is through, although octovis and dynamic_edt_3d are having problems. Need to look into it.

**Comment by wjwwood** [comment]
Thanks :+1:. I think `octomap` is the major blocker for other packages.

**Comment by ahornung** [comment]
I was able to fix a QGviewer linking problem in `devel` (https://github.com/OctoMap/octomap/commit/ed78af28e2398f5f5115ac5384767b24d0681153), the library names changed in Wily and Xenial. The prereleases and devel builds now succeed.

Unfortunately the[ octomap binary package](http://packages.ros.org/ros-shadow-fixed/ubuntu/pool/main/r/ros-kinetic-octomap/ros-kinetic-octomap_1.7.1-0wily-20160324-152855-0700_amd64.deb) is faulty, the CMakeConfig contains the wrong library paths pointing to the build tree instead:
```
set(OCTOMAP_INCLUDE_DIRS "/opt/ros/kinetic/include")
set(OCTOMAP_LIBRARY_DIRS "/opt/ros/kinetic/lib")
 

# Set library names
set(OCTOMAP_LIBRARIES 
  "/tmp/binarydeb/ros-kinetic-octomap-1.7.1/lib/liboctomap.so"
  "/tmp/binarydeb/ros-kinetic-octomap-1.7.1/lib/liboctomath.so"
)
```

**Comment by ahornung** [comment]
Fix released in v1.7.2, on the way to the build farm.

**Comment by wjwwood** [comment]
Awesome! Thanks for looking into it so promptly. I'll close this :+1:
